smartsim/settings/settings.py

```diff
@@ -86,8 +86,10 @@ def create_batch_settings(
         "lsf": BsubBatchSettings,
     }
 
-    if launcher == "auto":
+    if launcher in ["auto", "dragon"]:
```

Comment:
can we avoid the confusing part here of nested `if` blocks?

```
if launcher in ["auto", "dragon"]:
  launcher = detect_launcher()

if launcher == "dragon":
  by_launcher["dragon"] = by_launcher["launcher"]
```

something feels really fishy about needing to detect a launcher when the value is not "auto". I think we're mis-using detect_launcher and potentially adding extra functionality.

If we want to verify the existence of a launcher, we might instead do something like:

```
detected_launcher = detect_launcher()
if launcher != detected_launcher:
  raise ValueError("the requested launcher could not be detected")
```

Reply:
Yeah, ditto on Experiment attributes being needed in the launcher if we do it that way. Let's see what happens with future refactors